wp/lib/bap_wp/tests/unit/test_cfg_path.ml

```diff
@@ -0,0 +1,310 @@
+open !Core
+open Bap.Std
+open OUnit2
+open Testing_utilities
+open Bap_wp
+
+module Pre = Precondition
+module Comp = Compare
+module Constr = Constraint
+module Env = Environment
+module Bool = Z3.Boolean
+module BV = Z3.BitVector
+module Out = Output
+module CP = Cfg_path
+module Node = Graphs.Ir.Node
+module Edge = Graphs.Ir.Edge
+module EdgeSet = Edge.Set
+module Graph = Graphs.Ir
+
+let code_of_sub sub env = Comp.{
+    env;
+    prog = sub;
+    mem = Memmap.empty;
+    code_addrs = Utils.Code_addrs.empty; }
+
+let string_of_edgeset (es : EdgeSet.t) : string =
+  EdgeSet.fold es ~init:"\n" ~f:(fun s e -> s ^ Graphs.Ir.Edge.to_string e ^ "\n")
+
+let edge_of_endpoints (src : Blk.t) (dst : Blk.t) : Edge.t =
```

Comment:
Change to edges_of_endpoints to handle the case of a "tight loop" between two nodes.